compute/cloud-client/src/main/java/compute/CreateInstance.java

```diff
@@ -28,78 +28,79 @@
+      // Instance creation requires at least one persistent disk and one network interface.
       AttachedDisk disk =
           AttachedDisk.newBuilder()
               .setBoot(true)
               .setAutoDelete(true)
-              .setType(Type.PERSISTENT)  // Can be set to either 'SCRATCH' or 'PERSISTENT'
+              .setType(Type.PERSISTENT)
               .setInitializeParams(
                   AttachedDiskInitializeParams.newBuilder().setSourceImage(sourceImage)
                       .setDiskSizeGb(diskSizeGb).build())
               .build();
 
-      // "default" network interface is created automatically for every project
-      NetworkInterface networkInterface = NetworkInterface.newBuilder().setName("default").build();
+      // Use the network interface provided in the networkName argument.
+      NetworkInterface networkInterface = NetworkInterface.newBuilder().setName(networkName)
+          .build();
 
-      // Bind machine_name, machine_type, disk and network interface to an instance
+      // Bind `instanceName`, `machineType`, `disk`, and `networkInterface` to an instance.
       Instance instanceResource =
           Instance.newBuilder()
-              .setName(machineName)
+              .setName(instanceName)
               .setMachineType(machineType)
               .addDisks(disk)
               .addNetworkInterfaces(networkInterface)
               .build();
 
-      System.out.println(String.format("Creating instance: %s at %s ", machineName, zone));
-      // Inserting the instance in the specified project and zone
+      System.out.println(String.format("Creating instance: %s at %s ", instanceName, zone));
+      // Insert the instance in the specified project and zone.
       Operation response = instancesClient.insert(project, zone, instanceResource);
 
-      ZoneOperationsClient zoneOperationsClient = ZoneOperationsClient.create();
-      // waits for the delete operation to complete
-      // timeout is set at 180000 or 3 minutes
-      // the operation status will be fetched once in every 3 seconds to avoid spamming the api
-      long startTime = System.currentTimeMillis();
-      while (response.getStatus() == Status.RUNNING
-          && System.currentTimeMillis() - startTime < 180000) {
-        response = zoneOperationsClient.get(project, zone, response.getId());
-        TimeUnit.SECONDS.sleep(3);
+      if (response.getStatus() == Status.RUNNING) {
+        // Wait for the create operation to complete; default timeout is 2 mins
+        response = zoneOperationsClient.wait(project, zone, response.getId());
       }
 
       if (response.hasError()) {
         System.out.println("Instance creation failed ! ! " + response.getError());
         return;
       }
       System.out.println("####### Instance creation complete #######");
+
+    } catch (com.google.api.gax.rpc.UnknownException e) {
+      System.out.println("####### Instance creation complete #######");
     }
```

Comment:
You might wish to explain why this error is ok.  If I got an unknown exception, I'd normally be concerned.

Reply:
This is a temporary workaround for LRO. SocketTimeoutException is masked as UnknownException while creating an instance. Hence, even when the instance creation process is running at the background UnknownException is being thrown.

Reply:
You might wish to add a comment about that - especially since it's a work around.